### DISCUSSION TITLE
Need some help creating a Button with click Event

### DISCUSSION THREAD

**Discussion by Lahirutech** [GrapesJS/grapesjs#3501] [Q&A]
I want to create a button with a click event to call a specific URL(API) 
looking to extend the button so the URL can be added by typing through a text panel or something

How to add a click event to a component we add as a block in the block manager.
Is there a separate attribute to add.
```javascript
 {
                id: 'section', // id is mandatory
                label: '<b>Section</b>', // You can use HTML/SVG inside labels
                attributes: { class: 'gjs-block-section' },
                content: `<section>
              <h1>This is a simple title gjs-block-section</h1>
              <div>This is just a Lorem text: Lorem ipsum dolor sit amet</div>
            </section>`,
}
```
  Any idea on how to implement that functionality is highly appreciated 
  Thank you very much

**Comment by artf** [comment]
Maybe [custom traits](https://grapesjs.com/docs/modules/Traits.html) is what you're looking for
